tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,374 @@
+## Target customers
+<!-- Of the customers / personas listed in the Epic doc, what subset are we delivering this scenario to serve? -->
+**Infrastructure operators / administrators** - In enterprise applications team that use GitOps, operators are responsible for configuring and managing the git repos for infrastructure and application configuration. They configure GitOps policies as well as the Kubernetes manifests, Helm charts, etc. that are applied to the application and the Kubernetes cluster.
+
+**Application developers** - Responsible for designing, developing and maintaining application code. Developers use GitOps by checking files into their git repositories to configure GitOps settings to auto deploy and monitor their application and dependent infrastructure.
+
+## Existing customer problem
+<!-- <Write this in first person. You basically want to summarize what “I” as a customer am trying to accomplish, why the current experience is a problem and the impact it has on me, my team, my work and or biz, etc…. i.e. “When I try to do x aspect of cloud native app development, I have the following challenges / issues….<details>. Those issues result in <negative impact those challenges / issues have on your work and or business.> -->
+Since my organization uses GitOps to manage Kubernetes clusters and applications that run in those clusters, I need to remain consistent in infrastructure and application deployment even when using Radius. This means that I need to be able to apply Radius-centric concepts like Environments, Recipes, Applications, and Resources in a way that is consistent with my existing GitOps workflows. Without clear guidance on how to do this, I am unsure how to use both GitOps and Radius for a better together experience.
+
+## Desired customer experience outcome
+<!-- <Write this as an “I statement” that expresses the new capability from customer perspective … i.e. After this scenario is implemented “I can do, x, y, z, steps in cloud native app developer and seamlessly blah blah blah …. As a result <summarize positive impact on your work / business>  -->
+With the implementation of this scenario, I can deploy and manage Radius Environments, their associated Recipes, Applications, Credentials, and Resources via my existing GitOps (Flux and ArgoCD) toolsets. This means that I don't have to author pipelines or manually run Radius commands to deploy Radius-managed resources. Instead, I can leverage a git repo as the source of truth for my Radius-managed resources and use my existing GitOps tooling to deploy and manage those resources. This ensures that my experience with managing Radius resources is consistent with how I normally use GitOps to manage Kubernetes clusters and applications.
+
+> There is no new Radius specific user experience for this scenario. Customers are only interacting with Flux or ArgoCD. However, note that we're combining two different but existing user experiences (i.e. GitOps `git push` + Radius `*.bicep`).
```

Comment:
I think I'm saying a similar thing, but I think it would be clearer to say "The new experience is using Bicep with GitOps instead of YAML/KRM". You're using a lot of words to say that.

Reply:
updated